synapse/federation/transport/client.py

```diff
@@ -1317,15 +1317,22 @@ def __init__(self, room_version: RoomVersion, v1_api: bool):
             prefix + "auth_chain.item",
             use_float=True,
         )
-        self._coro_event = ijson.kvitems_coro(
+        # TODO Remove the unstable prefix when servers have updated.
+        self._coro_unstable_event = ijson.kvitems_coro(
             _event_parser(self._response.event_dict),
             prefix + "org.matrix.msc3083.v2.event",
             use_float=True,
         )
+        self._coro_event = ijson.kvitems_coro(
+            _event_parser(self._response.event_dict),
+            prefix + "event",
+            use_float=True,
+        )
```

Comment:
This means that the contents of both the `org.matrix.msc3083.v2.event` and `event` fields will stomp over each other. In a well behaving server they should be identical so this should be "OK". It might be possible to do something weird with this though.

It would be possible to keep these as two separate dictionaries and then do one of the following:

* Use `event` if it is non-empty, otherwise fallback to `org.matrix.msc3083.v2.event`
* Compare the two and ensure they are identical.

I'm a little concerned that this could be used to split-brain a room between different Synapse versions, maybe? But I don't really see how so I think I'm just being paranoid.

Reply:
> This means that the contents of both the `org.matrix.msc3083.v2.event` and `event` fields will stomp over each other.

Could you turn this into a code comment please? It was quite difficult for me to understand what's going on before I read this comment.